packages/cli/src/lib/snapshot/expandedPreviewer.ts

```diff
@@ -0,0 +1,219 @@
+    const previewLocalSlug = `${this.orgId}-${Date.now()}`;
+    const dirPath = resolve(
+      join(ExpandedPreviewer.previewDirectory, previewLocalSlug)
+    );
+    mkdirSync(dirPath, {
+      recursive: true,
+    });
+    const project = new Project(dirPath);
+    await this.initPreviewDirectory(project);
+    await this.applySnapshotToPreview(dirPath);
+    // TODO: Remove/move as tests progress.
+    if (Date.now() > 0) return;
+    cli.info(dedent`
+    
+    A Git repository representing the modification has been created here:
+    ${dirPath}
+    
+    `);
+  }
+
+  private deleteOldestPreviews() {
+    const getFilePath = (fileDirent: Dirent) =>
+      join(ExpandedPreviewer.previewDirectory, fileDirent.name);
+
+    const getEpochFromSnapshotDir = (dir: Dirent): number =>
+      parseInt(dir.name.match(/(?<=-)\d+$/)?.[0] ?? '0');
+
+    const allFiles = readdirSync(ExpandedPreviewer.previewDirectory, {
+      withFileTypes: true,
+    });
+    const dirs = allFiles
+      .filter((potentialDir) => potentialDir.isDirectory())
+      .sort(
+        (dirA, dirB) =>
+          getEpochFromSnapshotDir(dirA) - getEpochFromSnapshotDir(dirB)
+      );
+
+    while (dirs.length >= ExpandedPreviewer.previewHistorySize) {
+      rmSync(getFilePath(dirs.shift()!), {
+        recursive: true,
+        force: true,
+      });
+    }
+  }
+
+  private async initPreviewDirectory(project: Project) {
+    const beforeSnapshot = await this.getBeforeSnapshot();
+    await project.refresh(beforeSnapshot);
+    await this.initialPreviewCommit(project.pathToProject);
+  }
+
+  private async initialPreviewCommit(dirPath: string) {
+    await spawnProcess('git', ['init'], {cwd: dirPath, stdio: 'ignore'});
+    await spawnProcess('git', ['add', '.'], {cwd: dirPath, stdio: 'ignore'});
+    await spawnProcess('git', ['commit', `--message=${this.orgId} currently`], {
+      cwd: dirPath,
+      stdio: 'ignore',
+    });
+  }
+
+  private recursiveDirectoryDiff(
+    currentDir: string,
+    previewDir: string
+  ): string[] {
+    const files = readdirSync(previewDir, {withFileTypes: true});
+    const filePaths: string[] = [];
+    files.forEach((file) => {
+      if (file.isDirectory()) {
+        filePaths.push(
+          ...this.recursiveDirectoryDiff(
+            join(currentDir, file.name),
+            previewDir
+          )
+        );
+      }
+      if (file.isFile()) {
+        const previewFile = readJSONSync(join(previewDir, file.name));
+        const currentFilePath = join(currentDir, file.name);
+
+        if (!existsSync(currentFilePath)) {
+          writeJSONSync(join(currentDir, file.name), previewFile, {
+            spaces: 2,
+          });
+          return;
+        }
+
+        const currentFile = readJSONSync(currentFilePath);
+
+        const diffedJSON = this.buildDiffedJson(currentFile, previewFile);
+
+        writeJSONSync(join(currentDir, file.name), diffedJSON, {
+          spaces: 2,
+        });
+      }
+    });
```

Comment:
What happens if a file exist in `currentDir` but not in `previewDir`?

Reply:
For now, nothing.

🔜:tm: 
 - if shouldDelete is true, and the file ain't in previewDir, then we delete the file.
 - if shouldDelete is false, do nothing.

Reply:
- [ ] handle deleted files with shouldDelete flag